cinema/views.py

```diff
@@ -154,10 +170,30 @@
             queryset = queryset.filter(show_time__date=date)
 
         if movie_id_str:
-            queryset = queryset.filter(movie_id=int(movie_id_str))
+            try:
+                movie_id = int(movie_id_str)
+            except ValueError:
+                raise ValidationError(f"Invalid movie_id: {movie_id_str}")
```

Comment:
The code raises a 'ValidationError' but doesn't handle it. It's important to handle exceptions to prevent the application from crashing and to provide a useful error message to the user. You could catch this exception and return a response with a 400 status code and an error message.